apps/ex_wire/lib/ex_wire/config.ex

```diff
@@ -139,9 +140,15 @@ defmodule ExWire.Config do
     get_env(given_params, :p2p_version)
   end
 
-  @spec caps(Keyword.t()) :: [{String.t(), integer()}]
+  @spec caps(Keyword.t()) :: [Capability.t()]
   def caps(given_params \\ []) do
-    get_env(given_params, :caps)
+    caps = get_env(given_params, :caps)
```

Comment:
This (by its spec) shouldn't return nil. I'd just add a default value of the empty list `[]` here and then your map function below will work fine on that.